
### Examining racial discrimination in the US job market

#### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

#### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes.

#### Exercise
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Discuss statistical significance.

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet

****

In [2]:
import pandas as pd
import numpy as np
from scipy import stats

In [3]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [4]:
# number of callbacks for balck-sounding names
print "black callback %3.2F and white call back is %3.2F" % (sum(data[data.race=='b'].call), sum(data[data.race=='w'].call))

black callback 157.00 and white call back is 235.00


### 1. What test is appropriate for this problem? Does CLT apply?

We can use the two proportion sampling testing without pairing to test.  We can compare the proportion of black and white between the group with callback and the group without callback.  This two sampleing test should be non-pairing because the members in the callback group is not related to the members in the non-callback group.

we can use this test because it meets the criteria.  
- Sampling within group and between group seem to be random and less than 10% of population.  
- There are more than 10 successes and failures cases - Black callback total is 157 while white callback is 235. 

Based on the citeria above, the distribution is deemed to be normal and CLT should be applied for this test. 


 ### 2. What are the null and alternate hypotheses?

- The null hypthosis is that the proportion for race w in callback group - proportion for race w in non-callback group = 0
- The alternative hypothesis is that the proportion for race w in callback group - proportion for race w in non-callback group <> 0

### 3. Compute margin of error, confidence interval, and p-value.

Based on the computation in the cell below, the value for margin of error, confidence interval and p-value are:
- Margin of error: 0.051
- Confidence Interval: between 0.058 and 0.159
- p-value: 0.00002847 (two sides)


### 4.    Discuss statistical significance.

Since the p-value is lower than 0.05 significant level, I am rejecting the null hyperposis that the portion for race white with callback is the same as without callback.  

In [17]:
#to find proportion for white with and without callback
callback = data.race[data.call == 1]
nocallback = data.race[data.call == 0]
wpcallback = sum(callback == 'w') / float(callback.size)
wpnocallback = sum(nocallback == 'w') / float(nocallback.size)

#margin of error
se = np.sqrt(wpcallback*(1-wpcallback)/callback.size + wpnocallback*(1-wpnocallback)/nocallback.size)
#z score is 1.96 for 2 sides 95% confidence
me = stats.norm.ppf(0.975) * se
print "margin of error is %5.3F" % (me)

#confidence interval - point of estimate + or - margin of error
print "confidence interval is between %5.3F and %5.3F" % (wpcallback - wpnocallback - me, wpcallback - wpnocallback + me)

#p-value  = 
teststats = (wpcallback-wpnocallback) / se  #test statistic
print "p-value is %9.8F" % (stats.norm.sf(teststats)*2)



margin of error is 0.051
confidence interval is between 0.058 and 0.159
p-value is 0.00002847
